src/composable/types/Twap.ts

```diff
@@ -281,9 +305,14 @@ export class Twap extends ConditionalOrder<TwapData, TwapDataParams> {
     partSellAmount: BigNumber
     minPartLimit: BigNumber
   } {
-    return {
-      partSellAmount: o.sellAmount.div(o.n),
-      minPartLimit: o.buyAmount.div(o.n),
-    }
+    return !o.n.isZero()
+      ? {
+          partSellAmount: o.sellAmount.div(o.n),
+          minPartLimit: o.buyAmount.div(o.n),
+        }
+      : {
+          partSellAmount: BigNumber.from(0),
+          minPartLimit: BigNumber.from(0),
+        }
```

Comment:
I guess it's a protection for `division by zero` error

Reply:
yes, it is a protection from division by zero. 

I feel like, to make this generic, it would be nice to allow to create an invalid Twap class, and then validate it and get the error you have. 

This is why I prefer to not throw in the constructor, and let the SDK user decide.

This way, you can blindly create the order from the `staticInput` and make use of the methods of the class.
I can experiment later using this in WatchTower, and we can decide to move it back to the constructor if we see is a bad idea.